### PR TITLE
Update k8s to v1.30

### PR DESCRIPTION
**Pull request by karlkfi**
- k8s.io/kubernetes v1.30.1
- sigs.k8s.io/controller-runtime v0.18.4
- github.com/open-policy-agent/cert-controller v0.10.2+ 
  Using HEAD to pull in compatibility with controller-runtime v0.18.

Dependencies:
- https://github.com/GoogleContainerTools/kpt-config-sync/pull/1308
- https://github.com/GoogleContainerTools/kpt-config-sync/pull/1314
- https://github.com/GoogleContainerTools/kpt-config-sync/pull/1315

### CODE REVIEW

**Comment by karlkfi** [comment]
I don't really understand why, but go-licenses is now erroring with `Non go modules projects are no longer supported.` errors on a lot of go stdlib packages. AFAIK, that requirement was added in go-licenses v1.1.0. So I don't know what changed in this PR to trigger a new error from the same version of go-licenses.

File: go.mod [hunk 1]
```diff
@@ -1,9 +1,9 @@
 module kpt.dev/configsync
 
-go 1.22
+go 1.22.4
```
Comment (sdowell) [inline go.mod:3]: Is there a reason this pins to a patch version?

Comment (karlkfi) [reply go.mod:3]: It happens automatically now with Go 1.22 when you run `go mod tidy`. If I remove it, it comes back. Saw it happen in k8s upstream too.

Comment (karlkfi) [reply go.mod:3]: If I run `go mod tidy -go=1.22.0`, I can change it to  1.22.0, but If I try `go mod tidy -go=1.22` I get an error that `github.com/open-policy-agent/cert-controller@v0.10.2-0.20240531181455-2649f121ab97 requires go@1.22.0, but 1.22 is requested`

Comment (karlkfi) [reply go.mod:3]: Seems like if I run `go mod tidy -go=1.22.0` and then `go mod tidy`, it keeps the older version. So maybe that's the right move, to set the minimum requirement as low as possible.

Comment (sdowell) [reply go.mod]: Does this dictate which Go patch version is used when binaries are built? If so, we might want this to be the latest patch version. The golang builder image is currently set to `1.22.5`

Comment (karlkfi) [reply go.mod]: AFAIK, the version in go.mod is just the minimum version requirement to build with.

Comment (karlkfi) [reply go.mod]: > The minimum version of Go required by the current module.

https://go.dev/doc/modules/gomod-ref

Comment (karlkfi) [reply go.mod]: For comparison, upstream k8s is also using 1.22.0: https://github.com/kubernetes/kubernetes/blob/master/go.mod#L9

**Review by sdowell — APPROVED** [review]
/lgtm

**Pull request merged**